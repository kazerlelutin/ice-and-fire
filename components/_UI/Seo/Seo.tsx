import Head from 'next/head'

interface Props {
  readonly title?: string
}

export default function Seo({ title = 'Ice And Fire' }: Props) {
  return (
    <Head>
      <title>{title}</title>
      <meta name="description" content="It's my technical test !" />
      <link rel="icon" href="/favicon.ico" />
    </Head>
  )
}
