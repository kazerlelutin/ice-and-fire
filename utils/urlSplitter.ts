//for extract id 
export default function urlSplitter(url:string):string {
  const splitUrl = url.split('/')
  return splitUrl.at(-1)
}