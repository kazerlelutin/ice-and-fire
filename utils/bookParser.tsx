import { IapiBook } from "../types/apiBook.interface";
import { IBook } from "../types/book.interface";
import urlSplitter from "./urlSplitter";

export default function bookParser(book:IapiBook):IBook{

  return {
    id:urlSplitter(book.url),
    isbn: book.isbn,
    name: book.name,
    released: book.released,
    numberOfPages: book.numberOfPages,
    //merge all type of characters, just IDs
    characters: [...book.characters, ...book.povCharacters].reduce(
      (previousValue, currentValue) => {
        const 
        characterId = urlSplitter(currentValue),
        isExist = previousValue.find((id) => id === characterId)
        if (!isExist) previousValue.push(characterId)
        return previousValue
      },
      []
    ),
  }
}