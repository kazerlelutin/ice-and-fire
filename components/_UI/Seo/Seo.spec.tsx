import { afterEach, describe, expect, it, } from 'vitest'
import { cleanup } from '@testing-library/react'
import { render, screen, } from '../../../utils/test-utils'
import Seo from './Seo'

describe('<CommonLayout />', () => {

  afterEach(cleanup)
  
  it('Check snapshot', async () => {
    
    const { asFragment } = render(<Seo title='my title'/>)
    expect(asFragment()).toMatchSnapshot()
  })
})
