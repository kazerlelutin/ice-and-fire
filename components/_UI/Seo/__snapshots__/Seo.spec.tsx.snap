// Vitest Snapshot v1

exports[`<CommonLayout /> > Check snapshot 1`] = `<DocumentFragment />`;
